design_drafts/ros2_control_documentation.rst

```diff
@@ -0,0 +1,273 @@
+ <hardware>
+   <plugin>ros2_control_demo_hardware/ForceTorqueSensor1DHardware</plugin>
+   <param name="example_param_read_for_sec">0.43</param>
+ </hardware>
+ <sensor name="tcp_fts_sensor">
+   <state_interface name="force"/>
+   <param name="frame_id">rrbot_tcp</param>
+   <param name="min_force">-100</param>
+   <param name="max_force">100</param>
+ </sensor>
+</ros2_control>
+<ros2_control name="RRBotGripper" type="actuator">
+ <hardware>
+   <plugin>ros2_control_demo_hardware/PositionActuatorHardware</plugin>
+   <param name="example_param_write_for_sec">1.23</param>
+   <param name="example_param_read_for_sec">3</param>
+ </hardware>
+ <joint name="gripper_joint ">
+   <command_interface name="position">
+     <param name="min">0</param>
+     <param name="max">50</param>
+   </command_interface>
+   <state_interface name="position"/>
+   <state_interface name="velocity"/>
+ </joint>
+</ros2_control>
+
+
+Running the Framework for Your Robot
+------------------------------------
+To run the ros2_control framework, do the following.
+The example files can be found in the `ros2_control_demos`_ repository.
+
+#. Create a YAML  file with the configuration of the controller manager and controllers. (`Example for RRBot <https://github.com/ros-controls/ros2_control_demos/blob/master/ros2_control_demo_robot/controllers/rrbot_forward_controller_position.yaml>`_)
+#. Extend the robot's URDF description with needed ``<ros2_control>`` tags.
+   It is recommended to use macro files instead of pure URDF. (`Example for RRBot <https://github.com/ros-controls/ros2_control_demos/blob/master/ros2_control_demo_robot/description/rrbot_system_position_only.urdf.xacro>`_)
+#. Create a launch file to start the node with `Controller Manager`_.
+   You can use a default `ros2_control node`_ (recommended) or integrate the controller manager in your software stack.
+   (`Example launch file for RRBot <https://github.com/ros-controls/ros2_control_demos/blob/master/ros2_control_demo_robot/launch/rrbot_system_position_only.launch.py>`_)
+   
+Repositories
+============
+The ros2_control framework consist of the following repositories:
+
+ros2_control
+  The `ros2_control`_ repository implements the main interfaces and components of the framework mentioned in the previous sections.
+  
+ros2_controllers
+  The `ros2_controllers`_ repository implements widely used controllers, e.g., forward controller, joint trajectory controller, differential drive controller, etc.
+  
+ros2_control_demos
+  The `ros2_control_demos`_ repository provides examples of using the framework and templates for a smooth start with it.
+
+Differences to ros_control (ROS1)
+=================================
+
+Hardware Structures - classes
+-----------------------------
+
+The ros_control uses the ``RobotHW`` class as a rigid structure to handle any hardware.
+This makes it impossible to extend the existing robot with additional hardware, like sensors, actuators, and tools, without coding.
```

Comment:
`CombinedRobotHardware` is actually quite widely used, although an opt-in feature, not supported by default.